### PR TITLE
SWIG Python compilation with CMake >= 3.13

### PR DESCRIPTION
**Pull request by chapulina**
# 🦟 Bug fix

## Summary

The problem described in https://github.com/ignitionrobotics/ign-math/pull/216#issuecomment-900584411 turns out to be because of my CMake version, which is 3.21.1.

See [UseSWIG](https://cmake.org/cmake/help/latest/module/UseSWIG.html) as reference.

Version combinations that I'm trying to satisfy:

System | SWIG version | CMake version | Python version 
-- | -- | -- | --
Actions Bionic | 3.0.12 | 3.10.2 | 3.6.9
chapulina's Bionic | 3.0.12 | 3.21.1 | 3.6.9
Actions Focal | 4.0.1 | 3.16.3 | 3.8.10

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -110,17 +109,7 @@ if (PYTHONLIBS_FOUND)
...
-  # the library given to swig_add_library macro and the generated .so
-  if(NOT ${SWIG_VERSION} VERSION_GREATER 4.0.0)
-    set(SWIG_PY_LIB "_${SWIG_PY_LIB}")
-    set(SWIG_PY_LIB_OUTPUT "_${SWIG_PY_LIB_OUTPUT}")
-  endif()
-
-  set_target_properties(${SWIG_PY_LIB}
-    PROPERTIES
-      OUTPUT_NAME ${SWIG_PY_LIB_OUTPUT}
-  )
```
Comment (francocipollone) [inline src/CMakeLists.txt:123]: I thought it was related to the swig version as I commented there, but it seems that `if(NOT ${SWIG_VERSION} VERSION_GREATER 4.0.0)` should compare CMake version instead. In CI bionic is failing now.

**Comment by francocipollone** [comment]
Bionic CI is failing. I am checking this out locally right now

File: src/CMakeLists.txt (hunk 1 above)
Comment (francocipollone) [reply src/CMakeLists.txt]: I've tried this at e6f285d and I locally verified the behavior for:
`bionic -- cmake 10.x`
`bionic -- cmake 21.x`

Comment (chapulina) [reply src/CMakeLists.txt]: Great! Thanks for the quick fix!

**Review by francocipollone — APPROVED** [review]
Thanks @chapulina for finding and debugging this issue.
CI is happy now and I locally tried ubuntu with cmake 3.21.1

**Comment by chapulina** [comment]
Thanks for the fix, @francocipollone !

**Pull request merged**